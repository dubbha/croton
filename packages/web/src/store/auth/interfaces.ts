export interface SocialProfile {
  facebookId?: string;
  googleId?: string;
  pictureUrl: string;
}

export interface AuthState {
  isLoading: boolean;
  isAuthenticated: boolean;
  id: number | null;
  firstName: string | null;
  lastName: string | null;
  email: string | null;
  error: string | null;
  info: string | null;
  socialProfile: SocialProfile | null;
  isSignedInWithSocial: boolean;
}
