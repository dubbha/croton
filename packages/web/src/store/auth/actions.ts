export interface AuthState {
  isLoading: boolean;
  token: string | null;
  id: number | null;
  name: string | null;
  email: string | null;
  error: string | null;
}

export const AUTH_LOGIN = 'AUTH_LOGIN';
export const AUTH_LOGIN_SUCCESS = 'AUTH_LOGIN_SUCCESS';
export const AUTH_LOGIN_ERROR = 'AUTH_LOGIN_ERROR';

export const AUTH_REGISTER = 'AUTH_REGISTER';
export const AUTH_REGISTER_SUCCESS = 'AUTH_REGISTER_SUCCESS';
export const AUTH_REGISTER_ERROR = 'AUTH_REGISTER_ERROR';

export const AUTH_LOGOUT = 'AUTH_LOGOUT';

export interface AuthLogin {
  type: typeof AUTH_LOGIN;
  payload: { email: string; password: string };
}

export interface AuthLoginSuccess {
  type: typeof AUTH_LOGIN_SUCCESS;
  payload: { id: number; name: string; email: string; token: string };
}

export interface AuthLoginError {
  type: typeof AUTH_LOGIN_ERROR;
  payload: { error: string };
}

export interface AuthRegister {
  type: typeof AUTH_REGISTER;
  payload: {
    email: string,
    password: string,
    firstName: string,
    lastName: string
  };
}

export interface AuthRegisterSuccess {
  type: typeof AUTH_REGISTER_SUCCESS;
  payload: { id: number; name: string; email: string; token: string };
}

export interface AuthRegisterError {
  type: typeof AUTH_REGISTER_ERROR;
  payload: { error: string };
}

export interface AuthLogout { type: typeof AUTH_LOGOUT }

export type AuthActionTypes =
  | AuthLogin
  | AuthLoginSuccess
  | AuthLoginError
  | AuthRegister
  | AuthRegisterSuccess
  | AuthRegisterError
  | AuthLogout;
