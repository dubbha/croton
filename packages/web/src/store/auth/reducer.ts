import { LOCATION_CHANGE, LocationChangeAction } from 'connected-react-router';
import {
  AuthState,
  AuthActionTypes,
  AUTH_EMAIL_CONFIRM,
  AUTH_EMAIL_CONFIRM_ERROR,
  AUTH_EMAIL_CONFIRM_SUCCESS,
  AUTH_LOGIN,
  AUTH_LOGIN_SUCCESS,
  AUTH_LOGIN_ERROR,
  AUTH_REGISTER,
  AUTH_REGISTER_SUCCESS,
  AUTH_REGISTER_ERROR,
  AUTH_LOGOUT,
  AUTH_RESET_PASSWORD,
  AUTH_RESET_PASSWORD_SUCCESS,
  AUTH_RESET_PASSWORD_ERROR,
  AUTH_UPDATE_PASSWORD,
  AUTH_UPDATE_PASSWORD_SUCCESS,
  AUTH_UPDATE_PASSWORD_ERROR,
  AUTH_FACEBOOK,
  AUTH_FACEBOOK_SUCCESS,
  AUTH_FACEBOOK_ERROR,
  AUTH_UPDATE_PROFILE,
  AUTH_UPDATE_PROFILE_SUCCESS,
  AUTH_UPDATE_PROFILE_ERROR,
} from './actions';

export const initialState: AuthState = {
  isAuthenticated: false,
  id: null,
  firstName: null,
  lastName: null,
  email: null,
  isLoading: false,
  error: null,
  info: null,
};

export function authReducer(
  state = initialState,
  action: AuthActionTypes | LocationChangeAction
): AuthState {
  switch (action.type) {
    case AUTH_UPDATE_PASSWORD:
    case AUTH_RESET_PASSWORD:
    case AUTH_FACEBOOK:
    case AUTH_REGISTER:
    case AUTH_LOGIN:
    case AUTH_UPDATE_PROFILE:
      return {
        ...state,
        isLoading: true,
        error: null,
        info: null,
      };
    case AUTH_EMAIL_CONFIRM_SUCCESS:
    case AUTH_FACEBOOK_SUCCESS:
    case AUTH_LOGIN_SUCCESS: {
      const { id, firstName, lastName, email } = action.payload;
      return {
        ...state,
        isAuthenticated: true,
        id,
        firstName,
        lastName,
        email,
        isLoading: false,
      };
    }
    case AUTH_EMAIL_CONFIRM_ERROR:
    case AUTH_REGISTER_ERROR:
    case AUTH_LOGIN_ERROR:
    case AUTH_FACEBOOK_ERROR:
      return {
        ...state,
        isLoading: false,
        error: action.payload.error,
      };
    case AUTH_UPDATE_PASSWORD_SUCCESS:
    case AUTH_RESET_PASSWORD_SUCCESS:
    case AUTH_REGISTER_SUCCESS:
      return {
        ...state,
        isLoading: false,
        info: action.payload.info,
      };

    case AUTH_UPDATE_PROFILE_SUCCESS: {
      const { firstName, lastName, info } = action.payload;
      return {
        ...state,
        isLoading: false,
        info,
        firstName,
        lastName,
      };
    }

    case AUTH_UPDATE_PASSWORD_ERROR:
    case AUTH_RESET_PASSWORD_ERROR:
    case AUTH_UPDATE_PROFILE_ERROR:
      return {
        ...state,
        isLoading: false,
        error: action.payload.error,
        info: null,
      };

    case AUTH_EMAIL_CONFIRM:
      return {
        ...state,
        isLoading: true,
        error: null,
      };

    case AUTH_LOGOUT:
      return {
        ...initialState,
        isAuthenticated: false,
      };

    case LOCATION_CHANGE:
      return {
        ...state,
        isLoading: false,
        error: null,
        info: null,
      };
    default:
      return state;
  }
}
