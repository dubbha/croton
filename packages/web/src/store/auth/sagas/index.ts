export * from './authLogin.saga';
export * from './authLogout.saga';
export * from './authRegister.saga';
export * from './authResetPassword.saga';
export * from './authUpdatePassword.saga';
export * from './authEmailConfirm.saga';
export * from './authFacebook.saga';
export * from './authGoogle.saga';
