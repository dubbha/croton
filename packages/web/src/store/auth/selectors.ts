import { AppState } from '../interfaces';
import { AuthState } from './interfaces';

export const getAuth = (state: AppState): AuthState => state.auth;
