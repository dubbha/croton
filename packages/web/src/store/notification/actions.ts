export const NOTIFICATION_REGISTER = 'NOTIFICATION_REGISTER';
export const NOTIFICATION_REGISTER_SUCCESS = 'NOTIFICATION_REGISTER_SUCCESS';
export const NOTIFICATION_REGISTER_ERROR = 'NOTIFICATION_REGISTER_ERROR';

export interface NotificationRegister {
  type: typeof NOTIFICATION_REGISTER;
}

export interface NotificationRegisterSuccess {
  type: typeof NOTIFICATION_REGISTER_SUCCESS;
}

export interface NotificationRegisterError {
  type: typeof NOTIFICATION_REGISTER_ERROR;
  payload: { error: string };
}

export type NotificationActionTypes =
  | NotificationRegister
  | NotificationRegisterSuccess
  | NotificationRegisterError;
