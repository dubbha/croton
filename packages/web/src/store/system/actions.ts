export interface SystemState {
  loggedIn: boolean
  session: string
  userName: string
}

export const SYSTEM_UPDATE = 'SYSTEM_UPDATE';
export const SYSTEM_AUTH = 'SYSTEM_AUTH';
export const SYSTEM_REGISTER = 'SYSTEM_REGISTER'

export interface SystemUpdateAction {
  type: typeof SYSTEM_UPDATE
  payload: SystemState
}

export interface SystemAuthAction {
  type: typeof SYSTEM_AUTH,
  payload: {email: string, password: string},
}

export interface SystemRegisterAction {
  type: typeof SYSTEM_REGISTER,
  payload: {
    email: string,
    password: string,
    firstName: string,
    lastName: string
  },
}

export type SystemActionTypes =
  | SystemUpdateAction
  | SystemAuthAction
  | SystemRegisterAction;
