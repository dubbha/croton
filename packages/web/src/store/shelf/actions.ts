import { Actions } from 'constants/actions';
import { Shelf, Flower } from './interfaces';

export const SHELF_INVITE = 'SHELF_INVITE';
export const SHELF_INVITE_SUCCESS = 'SHELF_INVITE_SUCCESS';
export const SHELF_INVITE_ERROR = 'SHELF_INVITE_ERROR';
export const SHELF_INVITE_RESET = 'SHELF_INVITE_RESET';

export const SHELF_INVITE_ACCEPT = 'SHELF_INVITE_ACCEPT';
export const SHELF_INVITE_ACCEPT_SUCCESS = 'SHELF_INVITE_ACCEPT_SUCCESS';
export const SHELF_INVITE_ACCEPT_ERROR = 'SHELF_INVITE_ACCEPT_ERROR';

export const SHELF_DELETE_USER = 'SHELF_DELETE_USER';
export const SHELF_DELETE_USER_SUCCESS = 'SHELF_DELETE_USER_SUCCESS';
export const SHELF_DELETE_USER_ERROR = 'SHELF_DELETE_USER_ERROR';

export const SHELF_ADD_SHELF = 'SHELF_ADD_SHELF';
export const SHELF_ADD_SHELF_SUCCESS = 'SHELF_ADD_SHELF_SUCCESS';
export const SHELF_ADD_SHELF_ERROR = 'SHELF_ADD_SHELF_ERROR';

export const SHELF_EDIT_SHELF = 'SHELF_EDIT_SHELF';
export const SHELF_EDIT_SHELF_SUCCESS = 'SHELF_EDIT_SHELF_SUCCESS';
export const SHELF_EDIT_SHELF_ERROR = 'SHELF_EDIT_SHELF_ERROR';

export const SHELF_DELETE_SHELF = 'SHELF_DELETE_SHELF';
export const SHELF_DELETE_SHELF_SUCCESS = 'SHELF_DELETE_SHELF_SUCCESS';
export const SHELF_DELETE_SHELF_ERROR = 'SHELF_DELETE_SHELF_ERROR';

export const SHELF_GET_SHELVES = 'SHELF_GET_SHELVES';
export const SHELF_GET_SHELVES_SUCCESS = 'SHELF_GET_SHELVES_SUCCESS';
export const SHELF_GET_SHELVES_ERROR = 'SHELF_GET_SHELVES_ERROR';

export const SHELF_ADD_FLOWER = 'SHELF_ADD_FLOWER';
export const SHELF_ADD_FLOWER_SUCCESS = 'SHELF_ADD_FLOWER_SUCCESS';
export const SHELF_ADD_FLOWER_ERROR = 'SHELF_ADD_FLOWER_ERROR';

export const SHELF_EDIT_FLOWER = 'SHELF_EDIT_FLOWER';
export const SHELF_EDIT_FLOWER_SUCCESS = 'SHELF_EDIT_FLOWER_SUCCESS';
export const SHELF_EDIT_FLOWER_ERROR = 'SHELF_EDIT_FLOWER_ERROR';

export const SHELF_DELETE_FLOWER = 'SHELF_DELETE_FLOWER';
export const SHELF_DELETE_FLOWER_SUCCESS = 'SHELF_DELETE_FLOWER_SUCCESS';
export const SHELF_DELETE_FLOWER_ERROR = 'SHELF_DELETE_FLOWER_ERROR';

export const SHELF_GET_FLOWERS = 'SHELF_GET_FLOWERS';
export const SHELF_GET_FLOWERS_SUCCESS = 'SHELF_GET_FLOWERS_SUCCESS';
export const SHELF_GET_FLOWERS_ERROR = 'SHELF_GET_FLOWERS_ERROR';

export const SHELF_GET_FLOWER = 'SHELF_GET_FLOWER';
export const SHELF_GET_FLOWER_SUCCESS = 'SHELF_GET_FLOWER_SUCCESS';
export const SHELF_GET_FLOWER_ERROR = 'SHELF_GET_FLOWER_ERROR';

export const SHELF_ACTION = 'SHELF_ACTION';
export const SHELF_ACTION_SUCCESS = 'SHELF_ACTION_SUCCESS';
export const SHELF_ACTION_ERROR = 'SHELF_ACTION_ERROR';

export const SHELF_GET_LAST_ACTIONS = 'SHELF_GET_LAST_ACTIONS';
export const SHELF_GET_LAST_ACTIONS_SUCCESS = 'SHELF_GET_LAST_ACTIONS_SUCCESS';
export const SHELF_GET_LAST_ACTIONS_ERROR = 'SHELF_GET_LAST_ACTIONS_ERROR';

export const SHELF_RESET = 'SHELF_RESET';

export interface ShelfInvite {
  type: typeof SHELF_INVITE;
  payload: { userEmail: string; shelfId: number };
}

export interface ShelfInviteSuccess {
  type: typeof SHELF_INVITE_SUCCESS;
  payload: { info: string };
}

export interface ShelfInviteError {
  type: typeof SHELF_INVITE_ERROR;
  payload: { error: string };
}

export interface ShelfInviteAccept {
  type: typeof SHELF_INVITE_ACCEPT;
  payload: { shelfInvitationToken: string };
}

export interface ShelfInviteAcceptSuccess {
  type: typeof SHELF_INVITE_ACCEPT_SUCCESS;
  payload: { info: string };
}

export interface ShelfInviteAcceptError {
  type: typeof SHELF_INVITE_ACCEPT_ERROR;
  payload: { error: string };
}

export interface ShelfDeleteUser {
  type: typeof SHELF_DELETE_USER;
  payload: { shelfId: number, userId: number };
}

export interface ShelfDeleteUserSuccess {
  type: typeof SHELF_DELETE_USER_SUCCESS;
  payload: { info: string };
}

export interface ShelfDeleteUserError {
  type: typeof SHELF_DELETE_USER_ERROR;
  payload: { error: string };
}

export interface ShelfAddShelf {
  type: typeof SHELF_ADD_SHELF;
  payload: { name: string, description: string, location: string };
}

export interface ShelfAddShelfSuccess {
  type: typeof SHELF_ADD_SHELF_SUCCESS;
  payload: { info: string };
}

export interface ShelfAddShelfError {
  type: typeof SHELF_ADD_SHELF_ERROR;
  payload: { error: string };
}

export interface ShelfEditShelf {
  type: typeof SHELF_EDIT_SHELF;
  payload: { id: number, name: string, description: string, location: string };
}

export interface ShelfEditShelfSuccess {
  type: typeof SHELF_EDIT_SHELF_SUCCESS;
  payload: { info: string };
}

export interface ShelfEditShelfError {
  type: typeof SHELF_EDIT_SHELF_ERROR;
  payload: { error: string };
}

export interface ShelfDeleteShelf {
  type: typeof SHELF_DELETE_SHELF;
  payload: { id: number };
}

export interface ShelfDeleteShelfSuccess {
  type: typeof SHELF_DELETE_SHELF_SUCCESS;
  payload: { info: string };
}

export interface ShelfDeleteShelfError {
  type: typeof SHELF_DELETE_SHELF_ERROR;
  payload: { error: string };
}

export interface ShelfGetShelves {
  type: typeof SHELF_GET_SHELVES;
}

export interface ShelfGetShelvesSuccess {
  type: typeof SHELF_GET_SHELVES_SUCCESS;
  payload: { shelves: Shelf[] };
}

export interface ShelfGetShelvesError {
  type: typeof SHELF_GET_SHELVES_ERROR;
  payload: { error: string };
}

export interface ShelfAddFlower {
  type: typeof SHELF_ADD_FLOWER;
  payload: {
    shelfId: number,
    name: string,
    description: string,
    rrules: { [key in Actions]?: string }
  };
}

export interface ShelfAddFlowerSuccess {
  type: typeof SHELF_ADD_FLOWER_SUCCESS;
  payload: { info: string };
}

export interface ShelfAddFlowerError {
  type: typeof SHELF_ADD_FLOWER_ERROR;
  payload: { error: string };
}

export interface ShelfEditFlower {
  type: typeof SHELF_EDIT_FLOWER;
  payload: {
    id: number,
    shelfId: number,
    name: string,
    description: string,
    rrules: { [key in Actions]?: string },
  };
}

export interface ShelfEditFlowerSuccess {
  type: typeof SHELF_EDIT_FLOWER_SUCCESS;
  payload: { info: string };
}

export interface ShelfEditFlowerError {
  type: typeof SHELF_EDIT_FLOWER_ERROR;
  payload: { error: string };
}

export interface ShelfDeleteFlower {
  type: typeof SHELF_DELETE_FLOWER;
  payload: { id: number, shelfId: number };
}

export interface ShelfDeleteFlowerSuccess {
  type: typeof SHELF_DELETE_FLOWER_SUCCESS;
  payload: { info: string };
}

export interface ShelfDeleteFlowerError {
  type: typeof SHELF_DELETE_FLOWER_ERROR;
  payload: { error: string };
}

export interface ShelfGetFlowers {
  type: typeof SHELF_GET_FLOWERS;
  payload: { shelfId: number };
}

export interface ShelfGetFlowersSuccess {
  type: typeof SHELF_GET_FLOWERS_SUCCESS;
  payload: { flowers: Flower[] };
}

export interface ShelfGetFlowersError {
  type: typeof SHELF_GET_FLOWERS_ERROR;
  payload: { error: string };
}

export interface ShelfGetFlower {
  type: typeof SHELF_GET_FLOWER;
  payload: { id: number };
}

export interface ShelfGetFlowerSuccess {
  type: typeof SHELF_GET_FLOWER_SUCCESS;
  payload: { flower: Flower };
}

export interface ShelfGetFlowerError {
  type: typeof SHELF_GET_FLOWER_ERROR;
  payload: { error: string };
}

export interface ShelfAction {
  type: typeof SHELF_ACTION;
  payload: { action: Actions, flowerId: number, shelfId: number };
}

export interface ShelfActionSuccess {
  type: typeof SHELF_ACTION_SUCCESS;
  payload: { info: string };
}

export interface ShelfActionError {
  type: typeof SHELF_ACTION_ERROR;
  payload: { error: string };
}

export interface ShelfGetLastActions {
  type: typeof SHELF_GET_LAST_ACTIONS;
  payload: { flowerId: number, shelfId: number };
}

export interface ShelfGetLastActionsSuccess {
  type: typeof SHELF_GET_LAST_ACTIONS_SUCCESS;
  payload: {
    [key in Actions]: {
      timestamp: number;
      user: { id: number, firstName: string, lastName: string };
    }
  }
}

export interface ShelfGetLastActionsError {
  type: typeof SHELF_GET_LAST_ACTIONS_ERROR;
  payload: { error: string };
}

export interface ShelfReset {
  type: typeof SHELF_RESET;
}

export type ShelfActionTypes =
  | ShelfInvite
  | ShelfInviteSuccess
  | ShelfInviteError
  | ShelfInviteAccept
  | ShelfInviteAcceptSuccess
  | ShelfInviteAcceptError
  | ShelfDeleteUser
  | ShelfDeleteUserSuccess
  | ShelfDeleteUserError
  | ShelfAddShelf
  | ShelfAddShelfSuccess
  | ShelfAddShelfError
  | ShelfEditShelf
  | ShelfEditShelfSuccess
  | ShelfEditShelfError
  | ShelfDeleteShelf
  | ShelfDeleteShelfSuccess
  | ShelfDeleteShelfError
  | ShelfGetShelves
  | ShelfGetShelvesSuccess
  | ShelfGetShelvesError
  | ShelfAddFlower
  | ShelfAddFlowerSuccess
  | ShelfAddFlowerError
  | ShelfEditFlower
  | ShelfEditFlowerSuccess
  | ShelfEditFlowerError
  | ShelfDeleteFlower
  | ShelfDeleteFlowerSuccess
  | ShelfDeleteFlowerError
  | ShelfGetFlowers
  | ShelfGetFlowersSuccess
  | ShelfGetFlowersError
  | ShelfGetFlower
  | ShelfGetFlowerSuccess
  | ShelfGetFlowerError
  | ShelfAction
  | ShelfActionSuccess
  | ShelfActionError
  | ShelfGetLastActions
  | ShelfGetLastActionsSuccess
  | ShelfGetLastActionsError
  | ShelfReset;
