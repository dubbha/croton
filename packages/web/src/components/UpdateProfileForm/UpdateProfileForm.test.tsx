import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { AUTH_RESET_EMAIL, AUTH_UPDATE_PROFILE } from 'store/auth/actions';

describe('components/UpdateProfileForm', () => {
  const firstName = 'Chuck';
  const lastName = 'Norris';
  const email = 'bigbangkickedbyleg@hero.com';
  const error = 'Has been defeated by Bruce Lee';
  const info = 'Chuck Norris admits to having trained with Bruce Lee for approximately two years';

  const setUpMocks = ({ isSignedInWithSocial, info, error }) => {
    const mockDispatch = jest.fn();
    const mockUseSelector = jest.fn().mockReturnValue({
      isSignedInWithSocial,
      info,
      error,
      firstName,
      lastName,
      email,
    });

    const TEST_INFO_DISPLAY_TIME_MS = 0;

    jest.doMock('react-redux', () => ({
      useDispatch: () => mockDispatch,
      useSelector: mockUseSelector,
    }));

    jest.doMock('./constants', () => ({
      INFO_DISPLAY_TIME_MS: TEST_INFO_DISPLAY_TIME_MS,
    }));

    const { UpdateProfileForm } = require('./UpdateProfileForm');
    const { container, getByTestId } = render(<UpdateProfileForm />);

    return {
      container,
      mockDispatch,
      mockUseSelector,
      getByTestId,
    };
  };

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render successfully', () => {
    jest.isolateModules(() => {
      const { container } = setUpMocks({
        isSignedInWithSocial: false,
        info: null,
        error: null,
      });

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('should keep a placeholder for info or error', () => {
    jest.isolateModules(() => {
      const { getByTestId, container } = setUpMocks({
        isSignedInWithSocial: false,
        info: null,
        error: null,
      });

      expect(container).toContainElement(getByTestId('alertPlaceholder'));
    });
  });

  it('should display info if it is provided from auth', () => {
    jest.isolateModules(() => {
      const { container, getByTestId } = setUpMocks({
        isSignedInWithSocial: false,
        info,
        error: null,
      });

      expect(container.firstChild).toMatchSnapshot();
      expect(getByTestId('infoAlert').textContent).toBe(info);
    });
  });

  it('should display auth error', () => {
    jest.isolateModules(() => {
      const { container, getByTestId } = setUpMocks({
        isSignedInWithSocial: false,
        info: null,
        error,
      });

      expect(container.firstChild).toMatchSnapshot();
      expect(getByTestId('errorAlert').textContent).toBe(error);
    });
  });

  it('should pass first name, last name and email to the correspondent fields', () => {
    jest.isolateModules(() => {
      const { getByTestId } = setUpMocks({
        isSignedInWithSocial: false,
        info: null,
        error: null,
      });
      const testValues = [
        { dataTestId: 'updateProfileForm__firstName', value: firstName },
        { dataTestId: 'updateProfileForm__lastName', value: lastName },
        { dataTestId: 'updateProfileForm__changeEmail', value: email },
      ];

      testValues.forEach(({ dataTestId, value }) => {
        expect(getByTestId(dataTestId).getAttribute('value')).toBe(value);
      });
    });
  });

  it('should render first and last name fields as text inputs if a user is signed in via email', () => {
    jest.isolateModules(() => {
      const { getByTestId } = setUpMocks({
        isSignedInWithSocial: false,
        info: null,
        error: null,
      });
      const testValues = [
        'updateProfileForm__firstName',
        'updateProfileForm__lastName',
      ];

      testValues.forEach(dataTestId => {
        expect(getByTestId(dataTestId).getAttribute('type')).toBe('text');
      });
    });
  });

  it('should render the email field as enabled button is signed in via email', () => {
    jest.isolateModules(() => {
      const { getByTestId } = setUpMocks({
        isSignedInWithSocial: false,
        info: null,
        error: null,
      });

      expect(getByTestId('updateProfileForm__changeEmail')['disabled']).toBe(
        false,
      );
    });
  });

  it('should render first and last name fields as button inputs if a user is signed in via social', () => {
    jest.isolateModules(() => {
      const testValues = [
        'updateProfileForm__firstName',
        'updateProfileForm__lastName',
      ];
      const { getByTestId } = setUpMocks({
        isSignedInWithSocial: true,
        info: null,
        error: null,
      });

      testValues.forEach(dataTestId => {
        expect(getByTestId(dataTestId).getAttribute('type')).toBe('button');
      });
    });
  });

  it('should make the input fields disabled if a user is signed in via social', () => {
    jest.isolateModules(() => {
      const { getByTestId } = setUpMocks({
        isSignedInWithSocial: true,
        info: null,
        error: null,
      });
      const testValues = [
        'updateProfileForm__firstName',
        'updateProfileForm__lastName',
        'updateProfileForm__changeEmail',
      ];

      testValues.forEach(dataTestId => {
        expect(getByTestId(dataTestId)['disabled']).toBe(true);
      });
    });
  });

  it('should trigger reset email process after having clicked at the change email', () => {
    jest.isolateModules(() => {
      const { mockDispatch, getByTestId } = setUpMocks({
        isSignedInWithSocial: false,
        info: null,
        error: null,
      });

      fireEvent.click(getByTestId('updateProfileForm__changeEmail'));
      expect(mockDispatch).toHaveBeenCalledWith({ type: AUTH_RESET_EMAIL });
    });
  });

  it('should update the profile first name if it has been changed', () => {
    jest.isolateModules(() => {
      const { mockDispatch, getByTestId } = setUpMocks({
        isSignedInWithSocial: false,
        info: null,
        error: null,
      });

      const newFirstName = 'Clark';
      const firstNameField = getByTestId('updateProfileForm__firstName');

      fireEvent.change(firstNameField, {
        target: { value: newFirstName },
      });
      fireEvent.blur(firstNameField);

      expect(mockDispatch).toHaveBeenCalledWith({
        type: AUTH_UPDATE_PROFILE,
        payload: { firstName: newFirstName, lastName },
      });
    });
  });

  it('should not update the profile first name if the same value has been input by a user', () => {
    jest.isolateModules(() => {
      const { mockDispatch, getByTestId } = setUpMocks({
        isSignedInWithSocial: false,
        info: null,
        error: null,
      });

      const firstNameField = getByTestId('updateProfileForm__firstName');

      fireEvent.change(firstNameField, {
        target: { value: firstName },
      });
      fireEvent.blur(firstNameField);

      expect(mockDispatch).toHaveBeenCalledTimes(0);
    });
  });

  it('should update the profile last name if it has been changed', () => {
    jest.isolateModules(() => {
      const { mockDispatch, getByTestId } = setUpMocks({
        isSignedInWithSocial: false,
        info: null,
        error: null,
      });

      const newLastName = 'Kent';
      const lastNameField = getByTestId('updateProfileForm__lastName');

      fireEvent.change(lastNameField, {
        target: { value: newLastName },
      });
      fireEvent.blur(lastNameField);

      expect(mockDispatch).toHaveBeenCalledWith({
        type: AUTH_UPDATE_PROFILE,
        payload: { firstName, lastName: newLastName },
      });
    });
  });

  it('should not update the profile last name if the same value has been input by a user', () => {
    jest.isolateModules(() => {
      const { mockDispatch, getByTestId } = setUpMocks({
        isSignedInWithSocial: false,
        info: null,
        error: null,
      });

      const lastNameField = getByTestId('updateProfileForm__lastName');

      fireEvent.change(lastNameField, {
        target: { value: lastName },
      });
      fireEvent.blur(lastNameField);

      expect(mockDispatch).toHaveBeenCalledTimes(0);
    });
  });
});
