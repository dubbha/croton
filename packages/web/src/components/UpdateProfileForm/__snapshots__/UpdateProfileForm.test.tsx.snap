// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/UpdateProfileForm should display auth error 1`] = `
<form
  class=""
>
  <div
    class="fade alert alert-danger show"
    data-testid="errorAlert"
    role="alert"
  >
    Has been defeated by Bruce Lee
  </div>
  
  <div
    class="form-group"
  >
    <label
      class="form-label"
    >
      First Name
    </label>
    <input
      class="form-control"
      data-testid="updateProfileForm__firstName"
      placeholder="First Name"
      title="Update my first name"
      type="text"
      value="Chuck"
    />
    <label
      class="form-label"
    >
      Last Name
    </label>
    <input
      class="form-control"
      data-testid="updateProfileForm__lastName"
      placeholder="Last Name"
      title="Update my last name"
      type="text"
      value="Norris"
    />
    <label
      class="form-label"
    >
      Email
    </label>
    <input
      class="form-control"
      data-testid="updateProfileForm__changeEmail"
      placeholder="Email"
      title="Update my email"
      type="button"
      value="bigbangkickedbyleg@hero.com"
    />
  </div>
</form>
`;

exports[`components/UpdateProfileForm should display info if it is provided from auth 1`] = `
<form
  class=""
>
  <div
    class="fade alert alert-info show"
    data-testid="infoAlert"
    role="alert"
  >
    Chuck Norris admits to having trained with Bruce Lee for approximately two years
  </div>
  <div
    class="form-group"
  >
    <label
      class="form-label"
    >
      First Name
    </label>
    <input
      class="form-control"
      data-testid="updateProfileForm__firstName"
      placeholder="First Name"
      title="Update my first name"
      type="text"
      value="Chuck"
    />
    <label
      class="form-label"
    >
      Last Name
    </label>
    <input
      class="form-control"
      data-testid="updateProfileForm__lastName"
      placeholder="Last Name"
      title="Update my last name"
      type="text"
      value="Norris"
    />
    <label
      class="form-label"
    >
      Email
    </label>
    <input
      class="form-control"
      data-testid="updateProfileForm__changeEmail"
      placeholder="Email"
      title="Update my email"
      type="button"
      value="bigbangkickedbyleg@hero.com"
    />
  </div>
</form>
`;

exports[`components/UpdateProfileForm should render successfully 1`] = `
<form
  class=""
>
  
  <div
    data-testid="alertPlaceholder"
    style="height: 38px;"
  />
  <div
    class="form-group"
  >
    <label
      class="form-label"
    >
      First Name
    </label>
    <input
      class="form-control"
      data-testid="updateProfileForm__firstName"
      placeholder="First Name"
      title="Update my first name"
      type="text"
      value="Chuck"
    />
    <label
      class="form-label"
    >
      Last Name
    </label>
    <input
      class="form-control"
      data-testid="updateProfileForm__lastName"
      placeholder="Last Name"
      title="Update my last name"
      type="text"
      value="Norris"
    />
    <label
      class="form-label"
    >
      Email
    </label>
    <input
      class="form-control"
      data-testid="updateProfileForm__changeEmail"
      placeholder="Email"
      title="Update my email"
      type="button"
      value="bigbangkickedbyleg@hero.com"
    />
  </div>
</form>
`;
